js/src/common/utils/extractText.ts

```diff
@@ -1,15 +1,14 @@
+import Mithril from 'mithril';
+
 /**
  * Extract the text nodes from a virtual element.
- *
- * @param {VirtualElement} vdom
- * @return {String}
  */
-export default function extractText(vdom) {
+export default function extractText(vdom: Mithril.Children): string {
   if (vdom instanceof Array) {
     return vdom.map((element) => extractText(element)).join('');
   } else if (typeof vdom === 'object' && vdom !== null) {
-    return vdom.children ? extractText(vdom.children) : vdom.text;
+    return vdom.children ? extractText(vdom.children) : (vdom.text as string);
```

Comment:
Since text is `string | number | boolean`, maybe we should explicitly cast instead of asserting? If we're telling our type system that something is a string but it actually isn't, that could lead to trouble.